sphinx/domains/std.py

```diff
@@ -705,7 +653,10 @@ def resolve_xref(self, env, fromdocname, builder, typ, target, node, contnode):
         elif typ == 'option':
             resolver = self._resolve_option_xref
         elif typ == 'citation':
-            resolver = self._resolve_citation_xref
+            warnings.warn('pending_xref(domain=std, type=citation) is deprecated: %r' % node,
+                          RemovedInSphinx40Warning)
```

Comment:
I'm worry about this is breaking change or not.

Reply:
https://semver.org/ mentioned as:

1. MAJOR version when you make incompatible API changes,
2. MINOR version when you add functionality in a backwards-compatible manner, and
3. PATCH version when you make backwards-compatible bug fixes.

and I think 'deprecation warning' is not a breaking change so it is OK to be included in MINOR version.